src/backend/cdb/cdbmutate.c

```diff
@@ -75,8 +88,10 @@ typedef struct ApplyMotionState
 								 * plan_tree_walker/mutator */
 	int			nextMotionID;
 	int			sliceDepth;
-	bool		containMotionNodes;
 	HTAB	   *planid_subplans; /* hash table for InitPlanItem */
+
+	/* Context for ModifyTable to elide Explicit Redistribute Motion */
+	ModifyTableMotionState mt;
```

Comment:
Do we really need a new type which is used only once? I think it will be simpler to add all the 4 new fields to ApplyMotionState directly

Reply:
It's needed, because otherwise we'd need to include some prefix in the names of these members anyway. For example, `isChecking` is not a meaningful member name unless it's in a separate structure. I think the variant Georgy suggested is also ok.